guides/cpp-intro.md

```diff
@@ -0,0 +1,225 @@
+---
+layout: guide
+---
+
+# C++ introduction for the mbed developer
+
+The target audience for this guide is people who have not used C++ before and it will only cover the parts of C++ that is relevant for simple mbed programming.
+Understanding of programming languages in general is expected.
+
+# This code example
+
+This guide will use the following code example to explain how to use C++ with mbed.
```

Comment:
Perhaps explain what this code sample does, e.g. "It is a program called *blinky*, which flashes an LED on the development board on and off", or similar